packages/block-library/src/embed/util.js

```diff
@@ -178,3 +179,40 @@ export function fallback( url, onReplace ) {
 		createBlock( 'core/paragraph', { content: renderToString( link ) } )
 	);
 }
+
+/***
+ * Gets block attributes based on the preview and responsive state.
+ *
+ * @param {Object} preview The preview data.
+ * @param {string} title The block's title, e.g. Twitter.
+ * @param {Object} currentClassNames The block's current class names.
+ * @param {boolean} isResponsive Boolean indicating if the block supports responsive content.
+ * @param {boolean} allowResponsive Apply responsive classes to fixed size content.
+ * @return {Object} Attributes and values.
+ */
+export const getAttributesFromPreview = memoize( ( preview, title, currentClassNames, isResponsive, allowResponsive = true ) => {
```

Comment:
Isn't it kinda hard to follow that both the component method and the util function are named the same and then used throughout the code? I wonder if we could rename this one from util, `getMemoizedAttributesFromPreview` ?